pymc/sampling.py

```diff
@@ -253,6 +255,38 @@ def all_continuous(vars):
         return True
 
 
+def _get_seeds_per_chain(
+    random_state: RandomState,
+    chains: int,
+) -> Sequence[int]:
+    """Obtain new or validate specified integer seeds per chain"""
+
+    def _get_unique_seeds_per_chain(integers_fn):
+        seeds = []
+        while len(set(seeds)) != chains:
+            seeds = [int(seed) for seed in integers_fn(2**30, dtype=np.int64, size=chains)]
+        return seeds
+
+    if random_state is None or isinstance(random_state, int):
+        return _get_unique_seeds_per_chain(np.random.default_rng(random_state).integers)
+    if isinstance(random_state, np.random.Generator):
+        return _get_unique_seeds_per_chain(random_state.integers)
+    if isinstance(random_state, np.random.RandomState):
+        return _get_unique_seeds_per_chain(random_state.randint)
+
+    if isinstance(random_state, np.ndarray):
+        random_state = tuple(random_state)
+    elif not isinstance(random_state, (list, tuple)):
+        raise ValueError(f"The `seeds` must be array-like. Got {type(random_state)} instead.")
+
+    if len(random_state) != chains:
+        raise ValueError(
+            f"Number of seeds ({len(random_state)}) does not match the number of chains ({chains})."
+        )
+
+    return random_state
```

Comment:
What's the point of this function? To get different values to feed into the `numpy.random.SeedSequence` as the seeder's entropy? If yes, that's not the best approach to get independent random streams.
The best approach is to have a single entropy value for the `SeedSequence`, and spawn as many seeders as necessary. The resulting `spawn_key` should ensure that you get independent streams. From the [numpy docs](https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.SeedSequence.html#numpy.random.SeedSequence), it looks like we can feed a common entropy value to many `SeedSequence` instances but specify the `spawn_key`. So there must be a better way to do what we want, but I need to understand how this function is integrated to the rest of the `rngs` during sampling.

Reply:
I see, so we aren't passing seeds or anything like that to the steppers! We should open an issue to change that and have some way of providing independent streams given by `SeedSequence` to the steppers. We could use something like `SeedSequence(entropy=some_integer, spawn_key=(chain_idx, stepper_idx))`, which would guarantee independent streams of random numbers for each chain and stepper, but at the same time be deterministic given the entropy value, which would play the role of a single `random_seed` given to the `sample` function.

Anyway, given what you said about stepper relying on global seeds, this approach looks ok.

Reply:
Alternatively we can also just passa single generator or even seedsequence to each stepper and let it create/branch generators as they please. There's a vague issue about this here, we should update with more details: https://github.com/pymc-devs/pymc/issues/5093

Reply:
@lucianopaz I updated the docstrings of this function, added a direct test and made it so nothing is done if an integer is passed and a single chain is requested